sdk/storage/Azure.Storage.Common/src/StorageSharedKeyCredential.cs

```diff
@@ -74,5 +77,51 @@ private byte[] AccountKeyValue
         /// <returns>The signed message.</returns>
         protected static string ComputeSasSignature(StorageSharedKeyCredential credential, string message) =>
             credential.ComputeHMACSHA256(message);
+
+        /// <summary>
+        /// Parses a connection string for the storage account name and key
+        /// returns a <see cref="StorageSharedKeyCredential"/> created
+        /// from the connection string.
+        /// </summary>
+        /// <param name="connectionString">A valid connection string.</param>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="connectionString"/> is null or empty.</exception>
+        /// <exception cref="FormatException">Thrown if <paramref name="connectionString"/> is not a valid connection string.</exception>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="connectionString"/> cannot be parsed.</exception>
+        /// <returns>A <see cref="StorageConnectionString"/> object constructed from the values provided in the connection string.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static StorageSharedKeyCredential ParseConnectionString(string connectionString)
```

Comment:
We should remove this and use the existing [`StorageConnectionString.Parse`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/Shared/StorageConnectionString.cs#L239) which we're already doing in .ctors.  It'll properly handle escaping, etc.